### PR TITLE
docs(stories): Add some initial docs for the /stories/ Component Library

### PR DESCRIPTION
**Pull request by ryan953**
You should be able to visit the Component Library on any sentry instance, like on sentry.io: https://sentry.io/orgredirect/organizations/:orgslug/stories/

This is the first pass at some docs about it. Why we have it, how to get there, how to contribute and some ideas for the future!

### CODE REVIEW

File: src/docs/frontend/component-library.mdx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    // will render as <code>&lt;IconFire<br/>color="red"<br/>size="sm" \:gt;</code>
+    ```
+
+- [`<JSXProperty />`](https://github.com/getsentry/sentry/blob/master/static/app/components/stories/jsxProperty.tsx)
+    Render a formatted JSX property name & value.<br />
+    ```typescript
+    return <JSXProperty name="disabled" value={false} />;
+
+    // will render as `<code>size={false}</code>`
```
Comment (michellewzhang) [inline src/docs/frontend/component-library.mdx:95]:
```suggestion
    // will render as `<code>disabled={false}</code>`
```

File: src/docs/frontend/component-library.mdx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+Helper Components:
+
+<IconFire color="gray500" size="sm" />
+
+- [`<JSXNode />`](https://github.com/getsentry/sentry/blob/master/static/app/components/stories/jsxNode.tsx)<br />
+    Render a formatted JSX component name and some properties.<br />
+    ```typescript
+    return <JSXNode name="IconFire" props={{color: 'red400', size: 'sm'}} />;
+
+    // will render as <code>&lt;IconFire<br/>color="red"<br/>size="sm" \:gt;</code>
```
Comment (michellewzhang) [inline src/docs/frontend/component-library.mdx:87]: are the `&lt;` and `\:gt;` intentional?

File: src/docs/frontend/component-library.mdx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    By default uses `display:flex; overflow: hidden;` which can test responsive components well. Can also be set to `display: block; overflow: auto;`.<br />
+    ```typescript
+    <SizingWindow style={{height: '100px'}}>
+      <LoadingTriangle />
+    </SizingWindow>`
+    ```
+
+- [`<Matrix />`](https://github.com/getsentry/sentry/blob/master/static/app/components/stories/matrix.tsx)<br />
+    A helper to render multiple pairs of properties and values in a grid.
+    For example, we can render all button `priority` values against all possible `size` values which results in a 4x5 matrix of outputs.
```
Comment (michellewzhang) [inline src/docs/frontend/component-library.mdx:121]: might be more clear if you say something like "render 2 button `priority` values against 2 possible `size` values which results in a 2x2..." which matches the example below, or you could match the example to the text

**Comment by scttcper** [comment]
document the documentation

File: src/docs/frontend/component-library.mdx (hunk 2 above)
Comment (ryan953) [reply src/docs/frontend/component-library.mdx:87]: Yes. Trying to illustrate that we will output visible angle-braces to the screen. `&gt;` is the html escape code for that.

File: src/docs/frontend/component-library.mdx (hunk 3 above)
Comment (ryan953) [reply src/docs/frontend/component-library.mdx:121]:
```suggestion
    For example, we can render N values for one props, against M values for another prop. The example below compares 4 values for button `size` against 5 values for `priority`, resulting in a 20 example buttons in a grid.
```

File: src/docs/frontend/component-library.mdx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    ```typescript
+    <Matrix
+      render={() => Button}
+      propMatrix={{
+        priority: ['default', 'primary'],
+        size: ['md', 'sm'],
```
Comment (ryan953) [inline src/docs/frontend/component-library.mdx:127]:
```suggestion
        priority: ['default', 'primary', 'danger', 'link', undefined],
        size: ['md', 'sm', 'xs', 'zero'],
```

File: src/docs/frontend/component-library.mdx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  story('Default', () => <Button>Default Button</Button>);
+});
+```
+
+This template includes all the basics:
+
+1. The filename is `button.stories.tsx` and sits directly beside `button.tsx` which is the component being described.
+2. The file uses a default export, there is no need to name things when there is only one export.
+3. To create the `storyBook()` we're passing in the `Button` component directly as the first argument. This sets the title at the top of the page, and avoids typos.
+4. We've implemented a function that accepts `story` as it's argument. This is inspired by `jest` and the [`describe()`](https://jestjs.io/docs/api#describename-fn) function.
```
Comment (gggritso) [inline src/docs/frontend/component-library.mdx:49]:
```suggestion
4. We've implemented a function that accepts `story` as its argument. This is inspired by `jest` and the [`describe()`](https://jestjs.io/docs/api#describename-fn) function.
```

**Review by gggritso — APPROVED** [review]
B-e-a utiful

File: src/docs/frontend/component-library.mdx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+<Alert level="info">
+  To jump straight in visit <Link to="https://sentry.io/orgredirect/organizations/:orgslug/stories/"><code>/stories/</code></Link> on your existing Sentry Org/Install.
+</Alert>
+
+## What is a Component Library
+
+This component library is a (growing) collection of all the React components and hooks that are available within the getsentry/sentry codebase. 
+
+The library consists of the framework itself, which helps to make stories available & discoverable, as well as the story content. Each story is meant to describe how to use a React Component or hook; what are the properties, behaviors, and common or recommended combinations.
```
Comment (michellewzhang) [inline src/docs/frontend/component-library.mdx:13]:
```suggestion
The library consists of the framework itself, which helps to make stories available & discoverable, as well as the story content. Each story is meant to describe how to use a React Component or hook, as well as the properties, behaviors, and common or recommended combinations.
```

File: src/docs/frontend/component-library.mdx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+export default storyBook(Button, story => {
+  story('Default', () => <Button>Default Button</Button>);
+});
+```
+
+This template includes all the basics:
+
+1. The filename is `button.stories.tsx` and sits directly beside `button.tsx` which is the component being described.
+2. The file uses a default export, there is no need to name things when there is only one export.
```
Comment (michellewzhang) [inline src/docs/frontend/component-library.mdx:47]:
```suggestion
2. The file uses a default export; there is no need to name things when there is only one export.
```

File: src/docs/frontend/component-library.mdx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+Using this framework you get some nice things automatically:
+
+- Stories are printed in the same order they are written in the source file
+- Each `story()` callback can be async
+
+Some tips to remember:
+
+- Don't be afraid to import anything else you need to help illustrate use cases for your component. It's useful to leverage `<p>` and `<Fragment>` in most cases for example.
+- If you are demonstrating a component that relies on a global hook like `useOrganization()` or `usePageFilters()`, then you should wrap your story with those context providers manually.
+- Named exports, or a mix of named and default exports are all supported. This is more useful with [non-framework stories](#non-framework-stories).
```
Comment (michellewzhang) [inline src/docs/frontend/component-library.mdx:61]:
```suggestion
- Named exports, or a mix of named and default exports, are all supported. This is more useful with [non-framework stories](#non-framework-stories).
```

File: src/docs/frontend/component-library.mdx [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+- If you are testing a hook then you must pass a string as the first argument to `storyBook()` instead of the hook itself because.
```
Comment (michellewzhang) [inline src/docs/frontend/component-library.mdx:62]: because what? 😵

**Review by michellewzhang — APPROVED** [review]
gif

File: src/docs/frontend/component-library.mdx (hunk 9 above)
Comment (ryan953) [reply src/docs/frontend/component-library.mdx:62]: just because!

**Pull request merged**